qsiprep/utils/bids.py

```diff
@@ -206,14 +206,14 @@ def collect_data(bids_dir, participant_label, session_id=None, filters=None, bid
     }
     bids_filters = filters or {}
     for acq, entities in bids_filters.items():
+        queries[acq]['session'] = session_id or Query.OPTIONAL
         queries[acq].update(entities)
 
     subj_data = {
         dtype: sorted(
             layout.get(
                 return_type='file',
                 subject=participant_label,
-                session=session_id or Query.OPTIONAL,
```

Comment:
This was conflicting with `**query` when `query` had a "session" key.

Reply:
will this overwrite the bids filter file value with the command line arg? I'm ok with this behavior but we should warn that the bids filter file value is being overwritten

Reply:
No, it should override the session from the processing_list with the one from the filter instead.